### PR TITLE
Should handle composer dependency loop

### PR DESCRIPTION
**Pull request by vladim1**
There is a composer dependency loop described [here](https://github.com/sonata-project/SonataUserBundle/issues/724#issuecomment-282235225)
This PR should fix it

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -23,7 +23,7 @@
         "symfony/security-acl": "~2.8|~3.0",
         "symfony/console": "~2.8|~3.0",
         "sonata-project/core-bundle": "~2.3,>=2.3.10",
-        "sonata-project/admin-bundle": "~2.3|~2.4@dev",
+        "sonata-project/admin-bundle": "^3.0",
```
Comment (greg0ire) [inline composer.json]: Maybe do not forbid 2? `~2.3|~2.4@dev|^3.0`

File: composer.json [hunk 2]
```diff
@@ -23,7 +23,7 @@
         "symfony/security-acl": "~2.8|~3.0",
         "symfony/console": "~2.8|~3.0",
         "sonata-project/core-bundle": "~2.3,>=2.3.10",
-        "sonata-project/admin-bundle": "~2.3|~2.4@dev",
+        "sonata-project/admin-bundle": "~2.3|~2.4@dev|^3.0",
```
Comment (core23) [inline composer.json]: Please drop version 2.x, because it is unsupported.

**Comment by jordisala1991** [comment]
Doing a rebase of this branch with 3.x makes sense?

This branch is 182 commits behind 3.x

File: composer.json (hunk 1 above)
Comment (greg0ire) [inline composer.json]: Oh yeah you were right after all, sorry.

**Comment by jordisala1991** [comment]
I am not sure how to merge this with 3.x, It is a merge hell (and I think we cannot rebase in a PR).

**Comment by jordisala1991** [comment]
@vladim1 I created a PR to make this branch work: https://github.com/sonata-project/SonataUserBundle/pull/847

**Comment by jordisala1991** [comment]
This has been merged @vladim1 #847 . Can you check if this branch works for you now? Closing this because this change is no longer needed

**Pull request closed without merging**